### DISCUSSION TITLE
Apollo Plugin `willResolveField` event is not fired

### DISCUSSION THREAD

**Discussion by lkrzyzanek** [dotansimha/envelop#1523] [Q&A]
We use envelop (as part of graphql-mesh) with Apollo as custom server.

Apollo plugins implementing execute phase works correctly. Events like `requestDidStart` or `willSendResponse` are fired without any issue. (https://www.apollographql.com/docs/apollo-server/integrations/plugins/)

The problem is that `willResolveField` is not fired at all. The `executionDidStart` which is just before it is executed.
This breaks plugins which use this event e.g. `ApolloServerPluginCacheControl`. (https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server-core/src/plugin/cacheControl/index.ts#L133)

Is there any reason/limitation that causes this issue?

Thanks.
Libor

**Comment by n1ru4l** [comment]
Can you share a full example of this? To me it seems like because you are using an envelop provided schema, apollo does not have access to it.

**Reply by lkrzyzanek** [reply]
Yes you're right. We use GraphQL Mesh which uses envelop and we configure the apollo-server as server implementation.
Very similar how it's recommended in docs: https://www.the-guild.dev/graphql/mesh/docs/getting-started/customize-mesh-server#provide-a-standalone-server-implementation

What is not clear to me why some apollo plugin events are fired and others not.
